x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/decision/AutoscalingCapacity.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.autoscaling.decision;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Represents current/required capacity of a single tier.
+ */
+public class AutoscalingCapacity implements ToXContent, Writeable {
+
+    private final StorageAndMemory tier;
```

Comment:
I wonder if the name of this should be general from the beginning, something indicating resources like `AutoscalingResources`?

Reply:
To be clear, I'm not suggesting that it be a general container, it would still be only storage and memory today, but we'll have CPU later and then we'd rename this to `CpuStorageAndMemory`? That's what I mean by general, the name only.

Reply:
Yes, that is nicer, chose your name in 7bed89ea2c27042fd8d036dd0e8358533679a76c